editor/actions.js

```diff
@@ -285,6 +285,22 @@ export function removeBlocks( uids ) {
 }
 
 /**
+ * Returns an action object used in signalling that the blocks
+ * corresponding to the specified UID set are to be removed, and then
+ * another block given focus. It differs from removeBlocks in that it
```

Comment:
Do we really want a new action. Could we just use a flag?

Reply:
Or dispatch two actions?

Reply:
Yes, we shouldn't have two concurrent actions with hard-to-distinguish semantics.

In the new `DELETE_BLOCKS` effect, the removal itself is not conditional on state (only on payload, with the check for `blockUids.length`). Therefore, we should keep a single action, `REMOVE_BLOCKS`, and add a small effect for it that handles the new focus.

Now, the PR deals with two separate things:
- Fixing deletion in multi-block selection
- Focusing siblings after deletion

The former is a bug fix, the latter a UX improvement. I like it, but it begs the question: if we want to refocus a block after a multi-block deletion, are there any other cases (single-block deletion) where we _don't_ want to refocus a block? Should we prefer a consistent behavior, whether that means always refocusing or never refocusing? cc @jasmussen 

If still we want to make a distinction, then our `effects.REMOVE_BLOCKS` could look like:

```js
if ( blockUids.length > 1 ) {
  const blockToFocus = …;
  dispatch( focusBlock( blockToFocus.uid ) );
}
```

Reply:
> Do we really want a new action. Could we just use a flag?

I've generally found flags are much harder to reason about later, and don't scale at all --- but that's my experience. I'm happy to go with the majority.

> Or dispatch two actions?

I incorrectly assumed that for multi-actions, that effects is where they should go. I'm not realising that effects is for more complicated things. I don't think I'd seen many examples of where more than one action was dispatched in a component, so I thought that if you needed to do that, you put it in effects. My mistake.

> are there any other cases (single-block deletion) where we don't want to refocus a block?

My rationale for keeping the old behaviour and adding new behaviour is that you **never** want to enforce focus in an action. You might find that something starts removing blocks asynchronously, and it can not focus grab from where you are. We could do some other logic, like only focus another block if we know for sure the focus was in one of the blocks to be removed (because nothing about the remove semantics enforces that they were selected), but I would strongly advise against always refocusing a block on `remove` unless it's a bit smarter than just "always".

In regards to the `actions`, `effects` separation, I'd really welcome something fixing it with a commit, and then I can look at the diff and see by example what your preferred approach is.